src/quo/components/wallet/approval_label/view.cljs

```diff
@@ -0,0 +1,67 @@
+(ns quo.components.wallet.approval-label.view
+  (:require [oops.core :refer [oget]]
+            [quo.components.buttons.button.view :as button]
+            [quo.components.icon :as icon]
+            [quo.components.markdown.text :as text]
+            [quo.components.wallet.approval-label.schema :as approval-label.schema]
+            [quo.components.wallet.approval-label.style :as style]
+            [quo.foundations.colors :as colors]
+            quo.theme
+            [react-native.core :as rn]
+            [react-native.hole-view :as hole-view]
+            [schema.core :as schema]
+            [utils.i18n :as i18n]))
+
+(def ^:private status-icons
+  {:approve   :i/alert
+   :approving :i/pending-state
+   :approved  :i/check})
+
+(def ^:private status-message
+  {:approve   :t/approve-amount-symbol
+   :approving :t/approving-amount-symbol
+   :approved  :t/approved-amount-symbol})
+
+(defn- view-internal
+  [{:keys [status customization-color token-value token-symbol
+           container-style button-props]
+    :or   {customization-color :blue}}]
+  (let [theme                                 (quo.theme/use-theme)
+        [container-width set-container-width] (rn/use-state 0)
+        on-layout                             (rn/use-callback #(set-container-width
+                                                                 (oget % :nativeEvent :layout :width)))]
```

Comment:
can we avoid the use of oops, it tends to cause crashes a lot 😢 
I think at one point we were considering moving away from it completely

Reply:
I don't recall having experienced crashes due to `oops`. @J-Son89, do you have an example code to share that caused a crash?

Reply:
I'm not that familiar with the library, for now, I have removed the usage.

Reply:
Updated! It looks like I'm a little rusty on the guidelines. I will refresh 💪

Reply:
> Updated! It looks like I'm a little rusty on the guidelines. I will refresh 💪

💪🏼 Thanks Ajay!

Reply:
> can we avoid the use of oops, it tends to cause crashes a lot 😢 I think at one point we were considering moving away from it completely

@ilmotta I agree, we've had problems multiple times for a property not being in the object, it's weird, but the library throws an exception instead of returning `nil`.

BTW,  :eyes:  https://github.com/applied-science/js-interop